### DISCUSSION TITLE
allow columnsCustomName[x] to be a function

### DISCUSSION THREAD

**Discussion by mewalig** [Studio-42/elFinder#3351] [Ideas]
Hi,

I'd like my column names in list view to be dynamic. An easy way and backward-compatible way to do this, would be for me to provide a custom function in uioptions.cwd.listView.columnsCustomName[x], which would only require the following change in elFinder.js around line 4973:

```
		self.getColumnName = function(key) {
			// return columnNames[key] || self.i18n(key); // removed
			var cn = columnNames[key] || self.i18n(key); // added
			return cn && cn.constructor == Function ? cn() : cn; // added
		};
```

Thoughts?

**Comment by nao-pon** [comment]
@mewalig Sounds good! I will consider implementing it.

**Reply by nao-pon** [reply]
Draft:

```javascript
		/**
		 * Gets the column name of cwd list view
		 *
		 * @param      String  key     The key
		 * @return     String  The column name.
		 */
		self.getColumnName = function(key) {
			var res = columnNames[key] || self.i18n(key);
			return typeof res === 'function'? res() : res;
		};
```

**Reply by nao-pon** [reply]
I done at commit 6f9ce56601be0839e05c4ca5c05762a18128f890 . Thanks! 👍
